src/page-analysis/background/index.js

```diff
@@ -46,11 +44,44 @@ async function performPageAnalysis({pageId, tabId}) {
     await updatePageSearchIndex()
 }
 
-export default async function analysePage({page, tabId}) {
+/**
+ * Performs in-tab page analysis for a given page document, fetching and storing
+ * further page data such as text and metadata.
+ *
+ * @param {page} page The page document to save analysis data to.
+ * @param {tabId} string The ID of the tab to perform page data extraction in.
+ * @returns {page} The updated page document containing any extra data found in analysis.
+ */
+export async function analysePageInTab({page, tabId}) {
     // Wait until its DOM has loaded.
     await whenPageDOMLoaded({tabId}) // TODO: catch e.g. tab close.
-    await performPageAnalysis({pageId: page._id, tabId})
+
+    // Run page data fetching in content script in the tab.
+    const extractPageContent = remoteFunction('extractPageContent', {tabId})
+    // Run favicon fetching logic in content script in the tab.
+    const extractFavIcon = () => getFavIconFromTab({ tabId })
+
+    await performPageAnalysis({pageId: page._id, tabId, extractPageContent, extractFavIcon})
     // Get and return the page.
     page = revisePageFields(await db.get(page._id))
     return {page}
 }
+
+/**
+ * Performs background page analysis for a given page document, fetching and storing
+ * further page data such as text and metadata.
+ *
+ * @param {page} page The page document to save analysis data to.
+ * @param {url} string The URL pointing to the data source for page data extraction.
+ * @returns {page} The updated page document containing any extra data found in analysis.
+ */
+export async function analysePageInBackground({ page, url }) {
+    // Run page data fetching in background
+    const extractPageContent = () => fetchPageDataInBackground({ url })
+    const extractFavIcon = () => getFavIconFromUrl({ url })
+
+    await performPageAnalysis({ pageId: page._id, extractPageContent, extractFavIcon })
+    // Get and return the page.
+    const revisedPage = revisePageFields(await db.get(page._id))
+    return { page: revisedPage }
+}
```

Comment:
It may be a messy if we keep trying to use the same `performPageAnalysis` function for different scenarios by passing in each analysis function as an argument. Looks like some refactoring could make things cleaner; not sure which factoring to use, would need to move things around a bit to see what works. I would start with inlining `performPageAnalysis` into the two `analysePageIn...` functions, then again pulling out the pieces they share.